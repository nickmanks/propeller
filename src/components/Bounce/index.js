import React from 'react';
import styles from './index.scss';

const Bounce = ()=> (
  <div className={styles.bounce}>
    <div className={styles.one}></div>
    <div className={styles.two}></div>
    <div className={styles.three}></div>
  </div>
);

export default Bounce;
