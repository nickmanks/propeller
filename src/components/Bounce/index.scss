@import '#scss/colors';

.bounce {
  margin: 100px auto 0;
  text-align: center;
  width: 70px;

  > div {
    animation: bouncedelay 1.4s infinite ease-in-out both;
    background-color: $primary;
    border-radius: 100%;
    display: inline-block;
    height: 20px;
    width: 20px;
  }
}

.one {
  animation-delay: -0.32s !important;
}

.two {
  animation-delay: -0.16s !important;
}

@keyframes bouncedelay {
  0%,
  80%,
  100% {
    transform: scale(0);
  }

  40% {
    transform: scale(1);
  }
}
