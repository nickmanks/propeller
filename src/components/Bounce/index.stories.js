import React from 'react';
import Bounce from '.';
import {component} from '#src/testing/snapshots';

component('Bounce').when('standard', ()=> <Bounce />);
