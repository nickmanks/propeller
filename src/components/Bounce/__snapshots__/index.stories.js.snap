// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`standard 1`] = `
<div
  className="mock-bounce"
>
  <div
    className="mock-one"
  />
  <div
    className="mock-two"
  />
  <div
    className="mock-three"
  />
</div>
`;
